tools/jvm/src/main/scala/org/scalajs/core/tools/linker/package.scala

```diff
@@ -0,0 +1,47 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.core.tools
+
+import org.scalajs.core.tools.sem.Semantics
+import org.scalajs.core.tools.javascript.OutputMode
+
+import org.scalajs.core.tools.linker.frontend.{LinkerFrontend, ParIncOptimizer, IncOptimizer}
+import org.scalajs.core.tools.linker.backend.{LinkerBackend, BasicLinkerBackend}
+import org.scalajs.core.tools.linker.backend.closure.ClosureLinkerBackend
+
+package object linker {
+
+  implicit class LinkerJVM(val linker: Linker.type) extends AnyVal {
+    def linker(semantics: Semantics, outputMode: OutputMode,
+        withSourceMap: Boolean, parallel: Boolean, full: Boolean,
```

Comment:
I think `full` should be renamed into `useClosureCompiler`. It would de-couple fullOpt from Closure. In theory, in the new pipeline, nothing prevents to use GCC in fastOpt mode, or, a contrario, not to use GCC in fullOpt mode. The latter could be a perfectly reasonable thing to do ... for example when emitting to ES6!